pkg/kubectl/cmd/annotate.go

```diff
@@ -114,6 +114,7 @@ func NewCmdAnnotate(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().StringVar(&options.resourceVersion, "resource-version", "", "If non-empty, the annotation update will only succeed if this is the current resource-version for the object. Only valid when specifying a single resource.")
 	usage := "Filename, directory, or URL to a file identifying the resource to update the annotation"
 	kubectl.AddJsonFilenameFlag(cmd, &options.filenames, usage)
+	cmdutil.AddRecursiveFlags(cmd)
```

Comment:
Thanks for the input @liggitt. I modeled the `--recursive` flag after the way that the Validate flag was [defined](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/util/helpers.go#L329) & is being [used](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/create.go#L73).


I'll have to restructure the PR to use the model you've defined, so that I don't extend the pattern I misinterpreted.

I'm hoping to get to this sometime this week.